In [2]:
import os
import scipy
import numpy as np
import xarray as xr
import matplotlib.pyplot as plt
import matplotlib.patches as patches
import rasterio
import cv2
import scipy.io
import h5py
from skimage import io
from scipy.io import loadmat, savemat
from skimage.transform import resize
from hypso import Hypso1
from osgeo import gdal
import warnings
warnings.filterwarnings("ignore", category=FutureWarning)

In [ ]:
def split_and_save_Spat_patches(mat_image_path):
    # Load the .mat image
    mat_image = loadmat(mat_image_path)
    image_data = mat_image['data']

    # Get the dimensions of the image
    image_bands, image_height, image_width = image_data.shape

    # Define the patch size
    patch_size = 64

    # Randomly select a continuous 100 bands in each patch

    # Split the image into patches and save them in the same directory as the input image
    image_dir = os.path.join(os.path.dirname(mat_image_path), 'data')
    for i in range(0, image_height, patch_size):
        for j in range(0, image_width, patch_size):
            if i+patch_size <= image_height and j+patch_size <= image_width:
                start_band = np.random.randint(0, image_bands - 100)
                patch = image_data[start_band:start_band+100, i:i+patch_size, j:j+patch_size]
                patch_name = f'patch_{i}_{j}.mat'
                patch_path = os.path.join(image_dir, patch_name)
                # print(patch_path, patch.shape)
                savemat(patch_path, {'img': patch})

split_and_save_Spat_patches('./grizzly.mat')

In [30]:
!bash pre_train_Spat.sh

/home/lofty/miniconda3/envs/AI/lib/python3.12/site-packages/timm/optim/optim_factory.py:7: FutureWarning: Importing from timm.optim.optim_factory is deprecated, please import via timm.optim
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.optim", FutureWarning)
/home/lofty/miniconda3/envs/AI/lib/python3.12/site-packages/timm/models/layers/__init__.py:48: FutureWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", FutureWarning)
Not using distributed mode
[15:11:59.230828] job dir: /home/lofty/CODE/HyperSIGMA-fork/Pretrain
[15:11:59.230892] Namespace(batch_size=32,
epochs=10,
accum_iter=1,
model='spat_mae_b',
image_size=64,
mask_ratio=0.75,
norm_pix_loss=True,
use_ckpt='none',
weight_decay=0.05,
lr=None,
blr=0.00015,
min_lr=0.0,
warmup_epochs=1,
data_path='/home/lofty/CODE/HyperSIGMA-fork/Pretrain/data',
data_band='100bands',
in_ch

In [3]:
!bash pre_train_Spec.sh

/home/lofty/miniconda3/envs/AI/lib/python3.12/site-packages/timm/optim/optim_factory.py:7: FutureWarning: Importing from timm.optim.optim_factory is deprecated, please import via timm.optim
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.optim", FutureWarning)
/home/lofty/miniconda3/envs/AI/lib/python3.12/site-packages/timm/models/layers/__init__.py:48: FutureWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", FutureWarning)
usage: MAE pre-training [--batch_size BATCH_SIZE] [--epochs EPOCHS]
                        [--accum_iter ACCUM_ITER] [--model MODEL]
                        [--image_size IMAGE_SIZE] [--mask_ratio MASK_RATIO]
                        [--norm_pix_loss] [--use_ckpt USE_CKPT]
                        [--weight_decay WEIGHT_DECAY] [--lr LR] [--blr LR]
                        [--min_lr LR] [--warmup_epochs N]
   

/home/lofty/miniconda3/envs/AI/lib/python3.12/site-packages/timm/optim/optim_factory.py:7: FutureWarning: Importing from timm.optim.optim_factory is deprecated, please import via timm.optim
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.optim", FutureWarning)
/home/lofty/miniconda3/envs/AI/lib/python3.12/site-packages/timm/models/layers/__init__.py:48: FutureWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", FutureWarning)
Not using distributed mode
[15:11:59.230828] job dir: /home/lofty/CODE/HyperSIGMA-fork/Pretrain
[15:11:59.230892] Namespace(batch_size=32,
epochs=10,
accum_iter=1,
model='spat_mae_b',
image_size=64,
mask_ratio=0.75,
norm_pix_loss=True,
use_ckpt='none',
weight_decay=0.05,
lr=None,
blr=0.00015,
min_lr=0.0,
warmup_epochs=1,
data_path='/home/lofty/CODE/HyperSIGMA-fork/Pretrain/data',
data_band='100bands',
in_channels=100,
patch_size=8,
output_dir='/home/lofty/CODE/HyperSIGMA-fork/Pretrain/model',
log_dir='/home/lofty/CODE/HyperSIGMA-fork/Pretrain/logs',
device='cuda',
seed=0,
resume='',
start_epoch=0,
num_workers=4,
pin_mem=True,
world_size=1,
local_rank=0,
dist_on_itp=False,
dist_url='env://',
gpu_num=1,
tag=100,
port=None,
distributed=False)
[15:11:59.230948] ######### 0
[15:11:59.233313] /home/lofty/CODE/HyperSIGMA-fork/Pretrain
[15:11:59.233605] len_dataset 153
[15:11:59.233659] 10_0.75_0.00015_0.05_32
[15:11:59.233843] <util.datasets.HyperionDataset object at 0x7f09af6d8e90>
[15:11:59.233880] Sampler_train = <torch.utils.data.distributed.DistributedSampler object at 0x7f09af6b8e30>
[15:12:00.434848] Model = SpatMAE(
  (patch_embed): PatchEmbed(
    (proj): Conv2d(100, 768, kernel_size=(8, 8), stride=(8, 8))
  )
  (pos_drop): Dropout(p=0.0, inplace=False)
  (blocks): ModuleList(
    (0): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): Identity()
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (1): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.00909090880304575)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (2): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.0181818176060915)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (3): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.027272727340459824)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (4): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.036363635212183)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (5): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.045454543083906174)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (6): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.054545458406209946)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (7): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.06363636255264282)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (8): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.0727272778749466)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (9): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.08181818574666977)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (10): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.09090909361839294)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (11): Block(
      (norm1): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=768, out_features=2304, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=768, out_features=768, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.10000000149011612)
      (norm2): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
  )
  (norm): LayerNorm((768,), eps=1e-06, elementwise_affine=True)
  (decoder_embed): Linear(in_features=768, out_features=512, bias=True)
  (decoder_blocks): ModuleList(
    (0): Block(
      (norm1): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=512, out_features=1536, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=512, out_features=512, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): Identity()
      (norm2): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=512, out_features=2048, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=2048, out_features=512, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (1): Block(
      (norm1): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=512, out_features=1536, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=512, out_features=512, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.00909090880304575)
      (norm2): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=512, out_features=2048, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=2048, out_features=512, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (2): Block(
      (norm1): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=512, out_features=1536, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=512, out_features=512, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.0181818176060915)
      (norm2): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=512, out_features=2048, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=2048, out_features=512, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (3): Block(
      (norm1): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=512, out_features=1536, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=512, out_features=512, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.027272727340459824)
      (norm2): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=512, out_features=2048, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=2048, out_features=512, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (4): Block(
      (norm1): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=512, out_features=1536, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=512, out_features=512, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.036363635212183)
      (norm2): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=512, out_features=2048, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=2048, out_features=512, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (5): Block(
      (norm1): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=512, out_features=1536, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=512, out_features=512, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.045454543083906174)
      (norm2): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=512, out_features=2048, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=2048, out_features=512, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (6): Block(
      (norm1): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=512, out_features=1536, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=512, out_features=512, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.054545458406209946)
      (norm2): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=512, out_features=2048, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=2048, out_features=512, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
    (7): Block(
      (norm1): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (attn): Attention(
        (qkv): Linear(in_features=512, out_features=1536, bias=True)
        (attn_drop): Dropout(p=0.0, inplace=False)
        (proj): Linear(in_features=512, out_features=512, bias=True)
        (proj_drop): Dropout(p=0.0, inplace=False)
      )
      (drop_path): DropPath(p=0.06363636255264282)
      (norm2): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
      (mlp): Mlp(
        (fc1): Linear(in_features=512, out_features=2048, bias=True)
        (act): GELU(approximate='none')
        (fc2): Linear(in_features=2048, out_features=512, bias=True)
        (drop): Dropout(p=0.0, inplace=False)
      )
    )
  )
  (decoder_norm): LayerNorm((512,), eps=1e-06, elementwise_affine=True)
  (decoder_pred): Linear(in_features=512, out_features=6400, bias=True)
)
[15:12:00.435158] base lr: 1.50e-04
[15:12:00.435192] actual lr: 1.87e-05
[15:12:00.435221] accumulate grad iterations: 1
[15:12:00.435250] effective batch size: 32
[15:12:00.435710] AdamW (
Parameter Group 0
    amsgrad: False
    betas: (0.9, 0.95)
    capturable: False
    differentiable: False
    eps: 1e-08
    foreach: None
    fused: None
    lr: 1.875e-05
    maximize: False
    weight_decay: 0.05
)
/home/lofty/CODE/HyperSIGMA-fork/Pretrain/util/misc.py:269: FutureWarning: `torch.cuda.amp.GradScaler(args...)` is deprecated. Please use `torch.amp.GradScaler('cuda', args...)` instead.
  self._scaler = torch.cuda.amp.GradScaler()
[15:12:00.435944] Start training for 10 epochs
/home/lofty/CODE/HyperSIGMA-fork/Pretrain/engine_pretrain.py:47: FutureWarning: `torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.
  with torch.cuda.amp.autocast(enabled=False):
[15:12:01.592441] Epoch: [0]  [0/4]  eta: 0:00:04  lr: 0.000000  loss: 0.3366 (0.3366)  time: 1.1546  data: 0.2821  max mem: 2357
[2024-12-06 15:12:01,592 INFO misc.py line 159 18579] Epoch: [0]  [0/4]  eta: 0:00:04  lr: 0.000000  loss: 0.3366 (0.3366)  time: 1.1546  data: 0.2821  max mem: 2357
[15:12:01.886890] Epoch: [0]  [3/4]  eta: 0:00:00  lr: 0.000014  loss: 0.3217 (0.3234)  time: 0.3620  data: 0.0706  max mem: 2619
[2024-12-06 15:12:01,887 INFO misc.py line 159 18579] Epoch: [0]  [3/4]  eta: 0:00:00  lr: 0.000014  loss: 0.3217 (0.3234)  time: 0.3620  data: 0.0706  max mem: 2619
[15:12:01.901294] Epoch: [0] Total time: 0:00:01 (0.3661 s / it)
[15:12:01.901615] Averaged stats: lr: 0.000014  loss: 0.3217 (0.3234)
[15:12:03.789337] Epoch: [1]  [0/4]  eta: 0:00:01  lr: 0.000019  loss: 0.2782 (0.2782)  time: 0.3024  data: 0.2050  max mem: 2619
[2024-12-06 15:12:03,789 INFO misc.py line 159 18579] Epoch: [1]  [0/4]  eta: 0:00:01  lr: 0.000019  loss: 0.2782 (0.2782)  time: 0.3024  data: 0.2050  max mem: 2619
[15:12:04.085035] Epoch: [1]  [3/4]  eta: 0:00:00  lr: 0.000018  loss: 0.2509 (0.2576)  time: 0.1493  data: 0.0513  max mem: 2619
[2024-12-06 15:12:04,085 INFO misc.py line 159 18579] Epoch: [1]  [3/4]  eta: 0:00:00  lr: 0.000018  loss: 0.2509 (0.2576)  time: 0.1493  data: 0.0513  max mem: 2619
[15:12:04.101458] Epoch: [1] Total time: 0:00:00 (0.1537 s / it)
[15:12:04.101520] Averaged stats: lr: 0.000018  loss: 0.2509 (0.2576)
[15:12:04.320415] Epoch: [2]  [0/4]  eta: 0:00:00  lr: 0.000018  loss: 0.2301 (0.2301)  time: 0.2173  data: 0.1193  max mem: 2619
[2024-12-06 15:12:04,320 INFO misc.py line 159 18579] Epoch: [2]  [0/4]  eta: 0:00:00  lr: 0.000018  loss: 0.2301 (0.2301)  time: 0.2173  data: 0.1193  max mem: 2619
[15:12:04.605351] Epoch: [2]  [3/4]  eta: 0:00:00  lr: 0.000017  loss: 0.2131 (0.2185)  time: 0.1254  data: 0.0299  max mem: 2619
[2024-12-06 15:12:04,605 INFO misc.py line 159 18579] Epoch: [2]  [3/4]  eta: 0:00:00  lr: 0.000017  loss: 0.2131 (0.2185)  time: 0.1254  data: 0.0299  max mem: 2619
[15:12:04.622426] Epoch: [2] Total time: 0:00:00 (0.1300 s / it)
[15:12:04.622583] Averaged stats: lr: 0.000017  loss: 0.2131 (0.2185)
[15:12:04.879479] Epoch: [3]  [0/4]  eta: 0:00:01  lr: 0.000017  loss: 0.2009 (0.2009)  time: 0.2553  data: 0.1322  max mem: 2619
[2024-12-06 15:12:04,879 INFO misc.py line 159 18579] Epoch: [3]  [0/4]  eta: 0:00:01  lr: 0.000017  loss: 0.2009 (0.2009)  time: 0.2553  data: 0.1322  max mem: 2619
[15:12:05.160726] Epoch: [3]  [3/4]  eta: 0:00:00  lr: 0.000015  loss: 0.1908 (0.1933)  time: 0.1339  data: 0.0331  max mem: 2619
[2024-12-06 15:12:05,160 INFO misc.py line 159 18579] Epoch: [3]  [3/4]  eta: 0:00:00  lr: 0.000015  loss: 0.1908 (0.1933)  time: 0.1339  data: 0.0331  max mem: 2619
[15:12:05.176678] Epoch: [3] Total time: 0:00:00 (0.1382 s / it)
[15:12:05.176738] Averaged stats: lr: 0.000015  loss: 0.1908 (0.1933)
[15:12:05.387959] Epoch: [4]  [0/4]  eta: 0:00:00  lr: 0.000014  loss: 0.1809 (0.1809)  time: 0.2096  data: 0.1179  max mem: 2619
[2024-12-06 15:12:05,388 INFO misc.py line 159 18579] Epoch: [4]  [0/4]  eta: 0:00:00  lr: 0.000014  loss: 0.1809 (0.1809)  time: 0.2096  data: 0.1179  max mem: 2619
[15:12:05.686551] Epoch: [4]  [3/4]  eta: 0:00:00  lr: 0.000012  loss: 0.1718 (0.1746)  time: 0.1269  data: 0.0295  max mem: 2619
[2024-12-06 15:12:05,686 INFO misc.py line 159 18579] Epoch: [4]  [3/4]  eta: 0:00:00  lr: 0.000012  loss: 0.1718 (0.1746)  time: 0.1269  data: 0.0295  max mem: 2619
[15:12:05.701682] Epoch: [4] Total time: 0:00:00 (0.1309 s / it)
[15:12:05.701747] Averaged stats: lr: 0.000012  loss: 0.1718 (0.1746)
[15:12:05.938878] Epoch: [5]  [0/4]  eta: 0:00:00  lr: 0.000011  loss: 0.1656 (0.1656)  time: 0.2355  data: 0.1440  max mem: 2619
[2024-12-06 15:12:05,939 INFO misc.py line 159 18579] Epoch: [5]  [0/4]  eta: 0:00:00  lr: 0.000011  loss: 0.1656 (0.1656)  time: 0.2355  data: 0.1440  max mem: 2619
[15:12:06.223294] Epoch: [5]  [3/4]  eta: 0:00:00  lr: 0.000009  loss: 0.1593 (0.1611)  time: 0.1298  data: 0.0360  max mem: 2619
[2024-12-06 15:12:06,223 INFO misc.py line 159 18579] Epoch: [5]  [3/4]  eta: 0:00:00  lr: 0.000009  loss: 0.1593 (0.1611)  time: 0.1298  data: 0.0360  max mem: 2619
[15:12:06.240755] Epoch: [5] Total time: 0:00:00 (0.1344 s / it)
[15:12:06.240919] Averaged stats: lr: 0.000009  loss: 0.1593 (0.1611)
[15:12:06.478642] Epoch: [6]  [0/4]  eta: 0:00:00  lr: 0.000008  loss: 0.1561 (0.1561)  time: 0.2361  data: 0.1453  max mem: 2619
[2024-12-06 15:12:06,478 INFO misc.py line 159 18579] Epoch: [6]  [0/4]  eta: 0:00:00  lr: 0.000008  loss: 0.1561 (0.1561)  time: 0.2361  data: 0.1453  max mem: 2619
[15:12:06.742514] Epoch: [6]  [3/4]  eta: 0:00:00  lr: 0.000005  loss: 0.1510 (0.1527)  time: 0.1248  data: 0.0364  max mem: 2619
[2024-12-06 15:12:06,742 INFO misc.py line 159 18579] Epoch: [6]  [3/4]  eta: 0:00:00  lr: 0.000005  loss: 0.1510 (0.1527)  time: 0.1248  data: 0.0364  max mem: 2619
[15:12:06.759187] Epoch: [6] Total time: 0:00:00 (0.1292 s / it)
[15:12:06.759364] Averaged stats: lr: 0.000005  loss: 0.1510 (0.1527)
[15:12:06.958530] Epoch: [7]  [0/4]  eta: 0:00:00  lr: 0.000005  loss: 0.1493 (0.1493)  time: 0.1974  data: 0.1078  max mem: 2619
[2024-12-06 15:12:06,958 INFO misc.py line 159 18579] Epoch: [7]  [0/4]  eta: 0:00:00  lr: 0.000005  loss: 0.1493 (0.1493)  time: 0.1974  data: 0.1078  max mem: 2619
[15:12:07.235699] Epoch: [7]  [3/4]  eta: 0:00:00  lr: 0.000003  loss: 0.1463 (0.1472)  time: 0.1184  data: 0.0270  max mem: 2619
[2024-12-06 15:12:07,235 INFO misc.py line 159 18579] Epoch: [7]  [3/4]  eta: 0:00:00  lr: 0.000003  loss: 0.1463 (0.1472)  time: 0.1184  data: 0.0270  max mem: 2619
[15:12:07.256764] Epoch: [7] Total time: 0:00:00 (0.1240 s / it)
[15:12:07.256945] Averaged stats: lr: 0.000003  loss: 0.1463 (0.1472)
[15:12:07.502553] Epoch: [8]  [0/4]  eta: 0:00:00  lr: 0.000002  loss: 0.1436 (0.1436)  time: 0.2436  data: 0.1482  max mem: 2619
[2024-12-06 15:12:07,502 INFO misc.py line 159 18579] Epoch: [8]  [0/4]  eta: 0:00:00  lr: 0.000002  loss: 0.1436 (0.1436)  time: 0.2436  data: 0.1482  max mem: 2619
[15:12:07.787394] Epoch: [8]  [3/4]  eta: 0:00:00  lr: 0.000001  loss: 0.1436 (0.1436)  time: 0.1319  data: 0.0371  max mem: 2619
[2024-12-06 15:12:07,787 INFO misc.py line 159 18579] Epoch: [8]  [3/4]  eta: 0:00:00  lr: 0.000001  loss: 0.1436 (0.1436)  time: 0.1319  data: 0.0371  max mem: 2619
[15:12:07.798840] Epoch: [8] Total time: 0:00:00 (0.1351 s / it)
[15:12:07.799063] Averaged stats: lr: 0.000001  loss: 0.1436 (0.1436)
[15:12:08.038248] Epoch: [9]  [0/4]  eta: 0:00:00  lr: 0.000001  loss: 0.1423 (0.1423)  time: 0.2373  data: 0.1479  max mem: 2619
[2024-12-06 15:12:08,038 INFO misc.py line 159 18579] Epoch: [9]  [0/4]  eta: 0:00:00  lr: 0.000001  loss: 0.1423 (0.1423)  time: 0.2373  data: 0.1479  max mem: 2619
[15:12:08.308986] Epoch: [9]  [3/4]  eta: 0:00:00  lr: 0.000000  loss: 0.1427 (0.1428)  time: 0.1268  data: 0.0370  max mem: 2619
[2024-12-06 15:12:08,309 INFO misc.py line 159 18579] Epoch: [9]  [3/4]  eta: 0:00:00  lr: 0.000000  loss: 0.1427 (0.1428)  time: 0.1268  data: 0.0370  max mem: 2619
[15:12:08.324101] Epoch: [9] Total time: 0:00:00 (0.1309 s / it)
[15:12:08.324171] Averaged stats: lr: 0.000000  loss: 0.1427 (0.1428)
[15:12:09.960691] Training time 0:00:09
